internal/rbd/controllerserver.go

```diff
@@ -424,6 +430,19 @@ func (cs *ControllerServer) CreateVolume(
 		return nil, err
 	}
 
+	// Apply Qos parameters to rbd image.
```

Comment:
The CreateVolume function is already at the limit of the complexity (number of if/else and loop statements). Maybe it makes sense to move these two parts to the `createBackingImage()` function?

https://github.com/ceph/ceph-csi/actions/runs/12390057314/job/34584308943?pr=5016#step:3:841

Reply:
Yes, I full agree.